truffle/migrations/1_initial_migration.js

```diff
@@ -3,23 +3,18 @@ require('dotenv').config({ path: `${process.env.PWD}/.env` });
 const { waitForAccountToUnlock, extractEnvAccountAndPwd } = require('../test/utils');
 const Migrations = artifacts.require("./Migrations.sol");
 
-module.exports = (deployer) => {
+module.exports = async (deployer) => {
```

Comment:
This method cannot be `async` otherwise the main execution thread will keep going without waiting inside statements to completed

Reply:
yea you are right but for some bloody reason Truffle doesn't seem to wait even if this would be a blocking method or promise. They run things in paralel but screw it, I rewrote it using promises.

<img width="1153" alt="Screen Shot 2019-03-18 at 09 38 59" src="https://user-images.githubusercontent.com/3595242/54517338-afb46000-4961-11e9-8591-2bfec515f0b7.png">